### PR TITLE
Change position of directMemifServer in xconnectNSServer chain

### LINKED ISSUE #359: Position of directMemifServer in xconnectNSServer chain should be changed
# Expected Behavior
`directmemif.NewServer()`  should be called after `connect.NewServer()` to be able to connect client and endpoint, if they chose memif mechanism.
# Current Behavior
`excludedPrefixesServer`  was called before `connect.NewServer()`

### PR DESCRIPTION
**Pull request by tigrulya-exe**
Should fix #359 
1. Changed position of directMemifServer in xconnectNSServer chain. Now `directmemif.NewServer()` was called after `connectioncontextkernel.NewServer()`
2. Added `directmemif.removeXConnect` to remove XConnectPairs from vppConfig, if client and endpoint use directmemif.
2. Removed empty vppconfig.Interfaces check in `metrics.metricsServer` Request method.  
3. Added unit test to `xconnectNSServer`

### CODE REVIEW

File: pkg/networkservice/chains/xconnectns/server_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		require.NotNil(t, config)
+		vppConfig := config.VppConfig
+		require.NotNil(t, config)
+		require.Empty(t, vppConfig.Interfaces)
+		require.Empty(t, vppConfig.XconnectPairs)
+	}))
+
+	_, err = server.Request(ctx, request(t))
+	require.NoError(t, err)
+}
```
Comment (denis-tingajkin) [inline pkg/networkservice/chains/xconnectns/server_test.go:152]: Also, I think we need sure that socket files have been created

**Review by denis-tingajkin — COMMENTED** [review]
In general looks good.

**Comment by denis-tingajkin** [comment]
Can you also create an issue for this patch?

File: pkg/networkservice/metrics/server.go [hunk 2]
```diff
@@ -45,9 +46,7 @@ func (s *metricsServer) Request(ctx context.Context, request *networkservice.Net
 		return nil, errors.New("VPPAgent config is missing")
 	}
 	ifaces := conf.GetVppConfig().GetInterfaces()
-	if len(ifaces) == 0 {
-		return nil, errors.New("VPPAgent config should contain at least one interface")
-	}
+
```
Comment (denis-tingajkin) [inline pkg/networkservice/metrics/server.go:49]:
```suggestion
	if len(ifaces) == 0 {	
		next.Server(ctx).Request(ctx, request)
	}
```

Comment (tigrulya-exe) [reply pkg/networkservice/metrics/server.go:53]: Fixed.

File: pkg/networkservice/metrics/server.go [hunk 3]
```diff
@@ -44,10 +45,12 @@ func (s *metricsServer) Request(ctx context.Context, request *networkservice.Net
 	if conf == nil {
 		return nil, errors.New("VPPAgent config is missing")
 	}
+
 	ifaces := conf.GetVppConfig().GetInterfaces()
 	if len(ifaces) == 0 {
-		return nil, errors.New("VPPAgent config should contain at least one interface")
+		return next.Server(ctx).Request(ctx, request)
```
Comment (denis-tingajkin) [inline pkg/networkservice/metrics/server.go:51]: Let's add logging here to notify that the metrics server is not used.

Comment (tigrulya-exe) [reply pkg/networkservice/metrics/server.go:54]: Added.

File: pkg/networkservice/metrics/server.go [hunk 4]
```diff
@@ -48,6 +50,7 @@ func (s *metricsServer) Request(ctx context.Context, request *networkservice.Net
 
 	ifaces := conf.GetVppConfig().GetInterfaces()
 	if len(ifaces) == 0 {
+		logrus.Info("MetricsServer skipped: empty vppconfig interfaces")
```
Comment (denis-tingajkin) [inline pkg/networkservice/metrics/server.go:53]:
```suggestion
                log.Entry(ctx).Warn("vppconfig has no interfaces")
```

**Review by denis-tingajkin — APPROVED** [review]

**Pull request merged**